local_publisher.py

```diff
@@ -30,8 +30,19 @@ def configure(self, context):
         super().configure(context)
         self._publisher = NioPublisher(
             topic="{}.{}".format(self.local_identifier(), self.topic()))
-        self._publisher.open(on_connected=self.conn_on_connected,
-                             on_disconnected=self.conn_on_disconnected)
+
+        try:
+            self._publisher.open(on_connected=self.conn_on_connected,
+                                 on_disconnected=self.conn_on_disconnected)
+        except TypeError as e:
+            self.logger.warning(
```

Comment:
Rather than `format(str(e))` we should always use the python logging `exc_info` flag.

```python
self.logger.warning("Connecting to an outdated communication module, will not be able to determine connection status", exc_info=True)
```

Although, in this case I don't think we actually need information about the exception since we know what caused it. So I would propose removing any exception info whatsoever and just log the warning message.